pkg/destroy/aws/aws.go

```diff
@@ -520,6 +532,10 @@ func deleteEC2InternetGateway(client *ec2.EC2, id string, logger logrus.FieldLog
 
 	for _, gateway := range response.InternetGateways {
 		for _, vpc := range gateway.Attachments {
+			if vpc.VpcId == nil {
+				logger.WithField("gateway", id).Warn("gateway does not have a VPC ID")
```

Comment:
The only route to `deleteEC2InternetGateway` already sets the ID in a field [here][1], so we don't need the `WithField` at this point.  You only need `WithField` when you discover new information beyond what the caller gave for the resource you're deleting, or when you switch to look up and remove other resources which are dependencies of the resource you're deleting.

[1]: https://github.com/openshift/installer/blob/5dc2bc9f8b5be67dfef87331ff26f9f82fd09837/pkg/destroy/aws/aws.go#L407